src/Libraries/RevitNodes/Elements/FamilyInstance.cs

```diff
@@ -543,6 +544,41 @@ public FamilyInstance SetRotation(double degree)
             return this;
         }
 
+        ///// <summary>
+        ///// Transforms this FamilyInstance from a source CoordinateSystem to a new context CoordinateSystem.
+        ///// </summary>
+        ///// <param name="fromCoordinateSystem">Source CoordinatSystem</param>
+        ///// <param name="contextCoordinateSystem">Context CordinateSystem</param>
+        ///// <returns>Transformed Element</returns>
+        public FamilyInstance Transform(CoordinateSystem fromCoordinateSystem, CoordinateSystem contextCoordinateSystem)
```

Comment:
Seems this implementation can be applied to all kinds of model elements, right? If yes, I think it can be moved into Element Category.

Reply:
As mentioned above, the reason i am using FamilyInstance is because i need to check the elements Transform, i cannont seem to find a way to do this if the element is not a FamilyInstance? Maybe you have some ideas?

Reply:
I use below code to do move and rotate, works for Wall, Stairs, FamilyInstance. From the document of ElementTransformUtils Class, "A collection of utilities allowing transformation of elements (e.g. move, rotate, mirror and copy). ", I think it works for all kinds of element.
`
public void MoveElement()
{
	Reference refer = Application.ActiveUIDocument.Selection.PickObject(ObjectType.Element);
	Transaction tran = new Transaction(Document, "Move Element");
	tran.Start();
	ElementTransformUtils.MoveElement(Document, refer.ElementId, new XYZ(10, 10, 10));
	tran.Commit();
}

public void rotateElement()
{
	Reference refer = Application.ActiveUIDocument.Selection.PickObject(ObjectType.Element);
	Line zAxis = Line.CreateBound(new XYZ(0, 0, 0), new XYZ(0, 0, 1));
	Transaction tran = new Transaction(Document, "Move Element");
	tran.Start();
	ElementTransformUtils.RotateElement(Document, refer.ElementId, zAxis, 90);
	tran.Commit();
}
`

Reply:
What i need is to check the old `Transform` of the element before applying any new `Transform` to it. For example the `SetRotation` node in `FamilyInstance` does this

    var oldTransform = InternalGetTransform();
    double[] oldRotationAngles;
    TransformUtils.ExtractEularAnglesFromTransform(oldTransform, out oldRotationAngles);
    var newRotationAngle = degree * Math.PI / 180;

    if (!oldRotationAngles[0].AlmostEquals(newRotationAngle, 1.0e-6))

it gets the old transformation by doing `InternalGetTransformation()` which uses the `GetTransform`. Im guessing that the `SetRotation` node only works with `FamilyInstance` because of this? Which is also why i had to move this to the `FamilyInstance`

Reply:
OK, I see. And I also find Element.SetLocation, which use with FamilyInstance.SetLocation, can get the same effect as FamilyInstance.Transform.

Reply:
I suggest to remove the new node `FamilyInstance.Transform`.